xamarin/_Shared/App.xaml.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Xamarin.Forms;
+using System.IO;
+using System.Xml;
+using System;
+
+namespace RealmTemplateApp
+{
+    public partial class App : Application
+    {
+        private string appId;
+        public static Realms.Sync.App RealmApp;
+
+        public App()
+        {
+            InitializeComponent();
+        }
+
+        protected override void OnStart()
+        {
+            try
+            {
+                appId = GetAppId();
+                RealmApp = Realms.Sync.App.Create(appId);
+
+                var navPage = App.RealmApp.CurrentUser == null ?
+                    new NavigationPage(new LoginPage()) :
+                    new NavigationPage(new TaskPage());
+
+                NavigationPage.SetHasBackButton(navPage, false);
+                MainPage = navPage;
+            }
+            catch (NullReferenceException nre)
+            {
+                // A NullReferenceException occurs if:
+                // 1. the config file does not exist, or
+                // 2. the config does not contain an "app-id" element.
+
+                // If the app-id value is incorrect, we handle that
+                // in the Login page.
```

Comment:
Why does this catch and fail silently? Wouldn't we want a loud failure (perhaps re-throw or Swift `fatalError()`-equivalent containing the message)?